docker/autoware-universe/Dockerfile

```diff
@@ -53,14 +53,70 @@ RUN chmod 644 /etc/glvnd/egl_vendor.d/10_nvidia.json
 RUN echo "source /opt/ros/${ROS_DISTRO}/setup.bash" > /etc/bash.bashrc
 CMD ["/bin/bash"]
 
-FROM devel as prebuilt
+FROM devel as builder
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
 ## Build and change permission for runtime data conversion
 RUN source /opt/ros/"$ROS_DISTRO"/setup.bash \
   && colcon build --cmake-args -DCMAKE_BUILD_TYPE=Release \
   && find /autoware/install -type d -exec chmod 777 {} \;
 
+ARG CUDA_RUNTIME_BASE_IMAGE
+# hadolint ignore=DL3006
+FROM $CUDA_RUNTIME_BASE_IMAGE as prebuilt
+
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+ARG ROS_DISTRO
+ARG SETUP_ARGS
+
+## Install apt packages
+# hadolint ignore=DL3008
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -y install --no-install-recommends \
+  git \
+  ssh \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/*
+
+## Copy files
+COPY autoware.repos setup-dev-env.sh ansible-galaxy-requirements.yaml amd64.env arm64.env /autoware/
+COPY ansible/ /autoware/ansible/
+WORKDIR /autoware
+RUN ls /autoware
+
+## Add GitHub to known hosts for private repositories
+RUN mkdir -p ~/.ssh \
+  && ssh-keyscan github.com >> ~/.ssh/known_hosts
+
+## Set up runtime environment
+RUN --mount=type=ssh \
+  ./setup-dev-env.sh -y $SETUP_ARGS --no-cuda-drivers --no-dev --ros-install-base universe \
```

Comment:
I'm sorry for overlooking it, but I think it might be better to also add `install_devel == 'true'` to `ccache`, `pre_commit`, `ros2_dev_tools`.
https://github.com/autowarefoundation/autoware/blob/b64cfdb55c2106934b5262abd8667c65671ed597/ansible/playbooks/universe.yaml#L37

What do you think?

Reply:
Very good suggestion, I think these packages are also not required in runtime. I will add it.